examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py

```diff
@@ -41,6 +47,123 @@
 from dagster_airlift.migration_state import AirflowMigrationState
 
 
+@suppress_dagster_warnings
+def build_defs_from_airflow_instance(
+    *,
+    airflow_instance: AirflowInstance,
+    defs: Optional[Definitions] = None,
+    sensor_minimum_interval_seconds: int = DEFAULT_AIRFLOW_SENSOR_INTERVAL_SECONDS,
+) -> Definitions:
+    defs = defs or Definitions()
+    # Probably a temp workaround?
+    if DefinitionsLoadContext.context_is_set():
+        context = DefinitionsLoadContext.get()
+    else:
+        context = DefinitionsLoadContext(load_type=DefinitionsLoadType.INITIALIZATION)
+    metadata_key = f"{AIRFLOW_SOURCE_METADATA_KEY_PREFIX}/{airflow_instance.name}"
+    definitions_metadata = _get_or_create_definitions_metadata(
+        context, metadata_key, airflow_instance, defs
+    )
+    return definitions_from_metadata(
+        context, definitions_metadata, defs, airflow_instance, sensor_minimum_interval_seconds
+    )
+
```

Comment:
This is a bit messy, but we should be able to refactor it quickly over a lightweight, reusable pattern.

Reply:
yea esp now that sean's changes landed, i think this feels a lot better.